src/input_common/gcadapter/gc_adapter.cpp

```diff
@@ -0,0 +1,349 @@
+            adapter_thread_running = false; // error reading from adapter, stop reading.
+        } else {
+            for (int port = 0; port < pads.size(); port++) {
+                pads[port] = GetPadStatus(port, adapter_payload_copy);
+            }
+        }
+        for (int port = 0; port < pads.size(); port++) {
+            if (DeviceConnected(port) && configuring) {
+                if (pads[port].button != PAD_GET_ORIGIN) {
+                    pad_queue[port].Push(pads[port]);
+                }
+
+                // Accounting for a threshold here because of some controller variance
+                if (pads[port].stick_x > pads[port].MAIN_STICK_CENTER_X + pads[port].THRESHOLD ||
+                    pads[port].stick_x < pads[port].MAIN_STICK_CENTER_X - pads[port].THRESHOLD) {
+                    pads[port].axis = GCAdapter::PadAxes::StickX;
+                    pads[port].axis_value = pads[port].stick_x;
+                    pad_queue[port].Push(pads[port]);
+                }
+                if (pads[port].stick_y > pads[port].MAIN_STICK_CENTER_Y + pads[port].THRESHOLD ||
+                    pads[port].stick_y < pads[port].MAIN_STICK_CENTER_Y - pads[port].THRESHOLD) {
+                    pads[port].axis = GCAdapter::PadAxes::StickY;
+                    pads[port].axis_value = pads[port].stick_y;
+                    pad_queue[port].Push(pads[port]);
+                }
+                if (pads[port].substick_x > pads[port].C_STICK_CENTER_X + pads[port].THRESHOLD ||
+                    pads[port].substick_x < pads[port].C_STICK_CENTER_X - pads[port].THRESHOLD) {
+                    pads[port].axis = GCAdapter::PadAxes::SubstickX;
+                    pads[port].axis_value = pads[port].substick_x;
+                    pad_queue[port].Push(pads[port]);
+                }
+                if (pads[port].substick_y > pads[port].C_STICK_CENTER_Y + pads[port].THRESHOLD ||
+                    pads[port].substick_y < pads[port].C_STICK_CENTER_Y - pads[port].THRESHOLD) {
+                    pads[port].axis = GCAdapter::PadAxes::SubstickY;
+                    pads[port].axis_value = pads[port].substick_y;
+                    pad_queue[port].Push(pads[port]);
+                }
+            }
+            PadToState(pads[port], state[port]);
+        }
+        std::this_thread::yield();
+    }
+}
+
+void Adapter::ScanThreadFunc() {
+    LOG_INFO(Input, "GC Adapter scanning thread started");
+
+    while (detect_thread_running) {
+        if (usb_adapter_handle == nullptr) {
+            std::lock_guard<std::mutex> lk(initialization_mutex);
+            Setup();
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+    }
+}
+
+void Adapter::StartScanThread() {
+    if (detect_thread_running) {
+        return;
+    }
+    if (!libusb_ctx) {
+        return;
+    }
+
+    detect_thread_running = true;
+    detect_thread = std::thread([=] { ScanThreadFunc(); });
```

Comment:
I think you can just do
```suggestion
    detect_thread = std::thread(&Adapter::ScanThreadFunc);
```

Reply:
Unfortunately not, for member methods I needed to do that weird way since the instance itself is calling its own method, it's not a static public method.